index.js

```diff
@@ -20,302 +41,596 @@ var prefs = Cc["@mozilla.org/preferences-service;1"].getService(Ci.nsIPrefServic
+    id:           tickerId + WIDGET_SUFFIX,
+    type:         "custom",
+    label:        tickerId,
+    removable:    true,
+    defaultArea:  CustomizableUI.AREA_NAVBAR,
+    /**
+     * Build widget iFrame content and attach listeners
+     **/
+    onBuild: function(aDocument) {
+      ticker.doc = aDocument;
+      var node = aDocument.createElement("toolbaritem");
+      var props = {
+        id:           this.id,
+        title:        "Bitcoin Price Ticker " + tickerId,
+        align:        "center",
+        label:        tickerTitle,
+        width:        "15px",
+        height:       10,
+        tooltiptext:  tickerTitle,
+        class:        "chromeclass-toolbar-additional panel-wide-item"
+      };
+      for (var p1 in props) {
+        node.setAttribute(p1, props[p1]);
+      }
 
-  function startAutoPriceUpdate(tickerId) {
-    if (tickers[tickerId].url && tickers[tickerId].jsonPath) {
-      var fetchURLDataWrapper = function() {
-        fetchURLData(tickerId, tickers[tickerId].url, tickers[tickerId].jsonPath);
+      var iframe = aDocument.createElement("iframe");
+      props = {
+        id:          tickerId + IFRAME_SUFFIX,
+        type:        "content",
+        src:         IFRAME_URL,
+        transparent: true
       };
-      fetchURLDataWrapper();
-      tickers[tickerId].timer = setInterval(fetchURLDataWrapper, 
-                                            (tickers[tickerId].updateInterval * 1000));
-    }
-  }
+      for (var p2 in props) {
+        iframe.setAttribute(p2, props[p2]);
+      }
+
+      node.appendChild(iframe);
+
+      var listener = {
+        onWidgetAdded: function(aWidgetId, aArea, aPosition) {
+          if (aWidgetId == this.id) {
+            dlog("onWidgetAdded for " + tickerId);
+            setTimeout(function() { // Allow the ticker's iFrame to be created
+              updateTickerRefreshIntervalForTicker(tickerId, true);
+            }, 500);
+          }
+        }.bind(this),
+        onCustomizeStart: function(aWindow) {
+          dlog("onCustomizeStart for " + tickerId);
+          setTimeout(function() { // Allow the ticker's iFrame to be created
+            updateTickerRefreshIntervalForTicker(tickerId, true);
+          }, 1000);
+        }.bind(this),
+        onWidgetMoved: function(aWidgetId, aArea, aOldPosition, aNewPosition) {
+          if (aWidgetId == this.id) {
+            dlog("onWidgetMoved for " + tickerId);
+            setTimeout(function () { // Wait for Customize tab to load
+              updateTickerRefreshInterval(true);
+            }, 500);
+          }
+        }.bind(this),
+        onWidgetDrag: function(aWidgetId, aArea) {
+          if (aWidgetId == this.id) {
+            dlog("onWidgetDrag for " + tickerId);
+            draggingWidget = true; // This triggers Ticker updates after DOM change
+          }
+        }.bind(this),
+        onWidgetAfterDOMChange: function(aNode, aNextNode, aContainer, aWasRemoval) {
```

Comment:
Line is too long.<br>'aWasRemoval' is defined but never used.<br>'aContainer' is defined but never used.<br>'aNextNode' is defined but never used.